raster/r.colors.out/prt_json.c

```diff
@@ -0,0 +1,271 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <grass/gis.h>
+#include <grass/glocale.h>
+#include <grass/parson.h>
+#include <grass/raster.h>
+
+#include "local_proto.h"
+
+#define RGB_STRING_LENGTH 20
+#define HEX_STRING_LENGTH 8
+#define HSV_STRING_LENGTH 30
+
+/*!
+   \brief Closes the file if it is not stdout.
+
+   \param fp file where to print color table rules
+ */
+static void close_file(FILE *fp)
+{
+    if (fp != stdout)
+        fclose(fp);
+}
+
+/*!
+   \brief Converts RGB color values to HSV format.
+
+   \param r red component of the RGB color
+   \param g green component of the RGB color
+   \param b blue component of the RGB color
+   \param[out] h pointer to store the calculated hue
+   \param[out] s pointer to store the calculated saturation
+   \param[out] v pointer to store the calculated value
+ */
+static void rgb_to_hsv(int r, int g, int b, float *h, float *s, float *v)
+{
+    float r_norm = (float)r / 255.0f;
+    float g_norm = (float)g / 255.0f;
+    float b_norm = (float)b / 255.0f;
+
+    float cmax = MAX(r_norm, MAX(g_norm, b_norm));
+    float cmin = MIN(r_norm, MIN(g_norm, b_norm));
+    float diff = cmax - cmin;
+
+    if (cmax == cmin) {
+        *h = 0;
+    }
+    else if (cmax == r_norm) {
+        *h = fmodf((60.0f * ((g_norm - b_norm) / diff) + 360.0f), 360.0f);
+    }
+    else if (cmax == g_norm) {
+        *h = fmodf((60.0f * ((b_norm - r_norm) / diff) + 120.0f), 360.0f);
+    }
+    else {
+        *h = fmodf((60.0f * ((r_norm - g_norm) / diff) + 240.0f), 360.0f);
+    }
+
+    if (cmax == 0) {
+        *s = 0;
+    }
+    else {
+        *s = (diff / cmax) * 100.0f;
+    }
+
+    *v = cmax * 100.0f;
+}
+
+/*!
+   \brief Writes color entry in JSON in specified clr_frmt.
+
+   \param r red component of RGB color
+   \param g green component of RGB color
+   \param b blue component of RGB color
+   \param clr_frmt color format to be used (RGB, HEX, HSV, XTERM).
+   \param color_object pointer to the JSON object
+ */
+static void set_color(int r, int g, int b, enum ColorFormat clr_frmt,
+                      JSON_Object *color_object)
+{
+    switch (clr_frmt) {
+    case RGB: {
+        char rgb_string[RGB_STRING_LENGTH];
+        snprintf(rgb_string, sizeof(rgb_string), "rgb(%d, %d, %d)", r, g, b);
+        json_object_set_string(color_object, "rgb", rgb_string);
+        break;
+    }
+
+    case HEX: {
+        char hex_string[HEX_STRING_LENGTH];
+        snprintf(hex_string, sizeof(hex_string), "#%02X%02X%02X", r, g, b);
+        json_object_set_string(color_object, "hex", hex_string);
+        break;
+    }
```

Comment:
If only one format can be outputted at a time, why the key changes? It could be always `color`. If the idea is that more than one format can be outputted in the future and we want to have this ready for that, it should be documented in the code.